app/controllers/reports_controller.rb

```diff
@@ -4,19 +4,15 @@ class ReportsController < ApplicationController
   before_filter :setup_search_options, :only => :index
 
   def index
-    report_authorized = resource_base.my_reports
-    @reports = report_authorized.search_for(params[:search], :order => params[:order]).paginate(:page => params[:page], :per_page => params[:per_page]).includes(:host)
+    @reports = resource_base.search_for(params[:search], :order => params[:order]).paginate(:page => params[:page], :per_page => params[:per_page]).includes(:host)
   end
 
   def show
     # are we searching for the last report?
     if params[:id] == "last"
-      conditions = { :host_id => Host.find(params[:host_id]).try(:id) } unless params[:host_id].blank?
+      conditions = { :host_id => Host.authorized(:view_hosts).find(params[:host_id]).try(:id) } if params[:host_id].present?
       params[:id] = resource_base.where(conditions).maximum(:id)
     end
-
-    return not_found if params[:id].blank?
```

Comment:
This line serves a useful purpose, it stops a double-render error when the ID is blank (e.g. %20).